packages/app/src/cli/services/generate/fetch-extension-specifications.ts

```diff
@@ -114,3 +121,23 @@ async function mergeLocalAndRemoteSpecs(
 
   return result
 }
+
+/**
+ * The AppManagementClient has relaxed requirements for some extension specifications
+ * This should be a temporary transform until we either use contracts for every spec, or all extensions are optional by default
+ */
+function appManagementSpecTransform(specs: RemoteAwareExtensionSpecification[]) {
+  // eslint-disable-next-line @typescript-eslint/no-explicit-any
+  return specs.map((spec: any) => {
+    if (spec.identifier === AppHomeSpecIdentifier) {
+      spec.schema.shape.application_url = spec.schema.shape.application_url.optional()
+      spec.schema.shape.embedded = spec.schema.shape.embedded.optional()
```

Comment:
We recently removed `.schema` from the spec type but its still fully set and used elsewhere, so if we went this approach we will have to add the type back, or find another approach